tests/test_crawl.py

```diff
@@ -91,12 +91,11 @@ def test_retry_conn_failed(self):
 
     @defer.inlineCallbacks
     def test_retry_dns_error(self):
-        with mock.patch('socket.gethostbyname',
-                        side_effect=socket.gaierror(-5, 'No address associated with hostname')):
-            crawler = self.runner.create_crawler(SimpleSpider)
-            with LogCapture() as l:
-                yield crawler.crawl("http://example.com/")
-            self._assert_retried(l)
+        crawler = self.runner.create_crawler(SimpleSpider)
+        with LogCapture() as l:
+            # try to fetch the homepage of a non-existent domain
+            yield crawler.crawl("http://dns.resolution.invalid/")
```

Comment:
According to @glyph in https://github.com/scrapy/scrapy/pull/2521#issuecomment-276429894 and the referenced RFC2606 , `.invalid` is a reserved TLD, so I hope it's fine to use here.